pingcap/tidb
# executor: forbid stale read table with tiflash

Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Problem Summary: Should ban tiflash stale read.


### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

```
mysql root@localhost:(none)> create table test.table1 (id int primary key);
mysql root@localhost:(none)> ALTER TABLE test.table1 SET TIFLASH REPLICA 1
mysql root@localhost:(none)> SELECT * FROM information_schema.tiflash_replica // ensure tiflash replica is up
mysql root@localhost:(none)> explain analyze select * from test.table1 as of timestamp NOW();
(1105, 'staleness requests requires tikv backend')
mysql root@localhost:(none)> explain analyze select count(*) from test.table1 as of timestamp NOW();
(1105, 'staleness requests requires tikv backend')
```

### Release note 

- No release note
